presentation/chart_view.py

```diff
@@ -0,0 +1,63 @@
+import urllib.parse as urlparse
+from datetime import datetime, timedelta, timezone
+from typing import List, Tuple
+from urllib.parse import parse_qs, unquote
+
+import dash_core_components as dcc
+import dash_html_components as html
+import plotly.graph_objs as go
+
+from domain.agents import Agents
+from domain.metric import Metric
+from domain.model import MeshResults
+from domain.model.mesh_results import HealthItem, MeshResults
+from domain.types import AgentID
+from presentation.health import Health
+
+
+class ChartView:
+    @staticmethod
+    def make_layout(from_agent, to_agent: AgentID, metric: Metric, results: MeshResults) -> html.Div:
+        # make chart title
+        agents = Agents(results)
+        from_alias = agents.get_alias(from_agent)
+        to_alias = agents.get_alias(to_agent)
+        title = f"{metric.value}: {from_alias} -> {to_alias}"
+
+        # make chart data
+        health = Health(from_agent, to_agent, results)
+        xdata = [ChartView._utc_to_localtime(item.time) for item in health.items]
+        ydata = ChartView._get_ydata(metric, health)
```

Comment:
With all my suggested changes in place, data extraction and preparation would look like this:

```
series = results.filter(from_agent, to_agent, metric)
fig = go.Figure(data=[go.Scatter(x=[utc_to_localtime(e[0]) for e in series], y=[e[1] for e in series])])
```

Reply:
Such one-liners are not really readable, IMO

Reply:
I agree that such long lines are too complicated. I personally like inlining variables and then doing line breaks like this:

```
fig = go.Figure(data=[go.Scatter(
    x=[utc_to_localtime(e[0]) for e in series],
    y=[e[1] for e in series],
)])
```

This is closer to functional style, when reader sees first _what_ is being done (go.Figure()) instead of _how_ it is being done (how xdata and ydata is calculated)

Reply:
The code was not intended to follow functional style; my intention here is to make it easier to read. And to breakpoint, when necessary

Reply:
Ok, that was just my opinion on what I find the most readable.